spec/browsers/deezer_user_agent_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'user_agent'
+
+shared_examples 'Deezer' do |version, platform, os, type|
+  it "returns 'Deezer' as its browser" do
+    expect(useragent.browser).to eq('Deezer')
+  end
+
+  it "returns '#{version}' as its version" do
+    expect(useragent.version).to eq(version)
+  end
+
+  it "returns '#{platform}' as its platform" do
+    expect(useragent.platform).to eq(platform)
+  end
+
+  it "returns '#{os}' as its operating system" do
+    expect(useragent.os).to eq(os)
+  end
+
+  if type == :mobile
+    it { expect(useragent).to be_mobile }
+    it { expect(useragent).not_to be_desktop }
+  else
```

Comment:
```suggestion
  elsif type == :desktop
```

Alternatively make this a boolean so there can't be a third state you mishandle.